core/app/models/spree/payment.rb

```diff
@@ -222,5 +229,42 @@ def update_order
         end
       end
 
+      # Necessary because some payment gateways will refuse payments with
```

Comment:
I think all of this except for create_eligible_credit_event method should be removed